### PR TITLE
DB-7876 Support STRIP() function

### PR DESCRIPTION
**Pull request by changli6**

### CODE REVIEW

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj [hunk 1]
```diff
@@ -2589,12 +2589,14 @@ TOKEN [IGNORE_CASE] :
...
 |	<ANY: "any">
 |	<ARE: "are">
+|	<ARRAY: "array">
 |	<AS: "as">
 |	<ASC: "asc">
 |	<ASSERTION: "assertion">
 |	<AT: "at">
 |	<AUTHORIZATION: "authorization">
 |	<AVG: "avg">
+|	<B: "b">
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj]: We should put "b", "L", "T" in the non-reserved keyword section.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/TernaryOperatorNode.java [hunk 2]
```diff
@@ -84,8 +84,9 @@
 	public static final int TIMESTAMPADD = 4;
 	public static final int TIMESTAMPDIFF = 5;
 	public static final int REPLACE = 6;
-	static final String[] TernaryOperators = {"trim", "LOCATE", "substring", "like", "TIMESTAMPADD", "TIMESTAMPDIFF", "replace"};
-	static final String[] TernaryMethodNames = {"ansiTrim", "locate", "substring", "like", "timestampAdd", "timestampDiff", "replace"};
+	public static final int STRIP = 7;
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/TernaryOperatorNode.java]: This change in this module does not seem necessary. I don't see a case where the operator type will be set to STRIP, as in sqlgrammer.jj, STRIP is eventually generated as a TrimOperatorNode. Please correct me if I missed something.

**Comment by changli6** [comment]
jenkins please test branch @cdh5.12.2

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj (hunk 1 above)
Comment (changli6) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj]: Moved them to non-reserved keywords section.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/TernaryOperatorNode.java (hunk 2 above)
Comment (changli6) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/TernaryOperatorNode.java]: Good catch. The STRIP function returns TrimOperatorNode node type with TernaryOperatorNode.TRIM operator, which is the same as TRIM. STRIP related stuff is unnecessary here.

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**